pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -422,24 +434,38 @@ func (m *rangefeedMuxer) receiveEventsFromNode(
 		// get stuck in the first place.
 		if timeutil.Now().Before(nextStuckCheck) {
 			if threshold := stuckThreshold(); threshold > 0 {
-				for _, a := range ms.purgeStuckStreams(threshold) {
-					if err := m.restartActiveRangeFeed(ctx, a, errRestartStuckRange); err != nil {
-						return err
-					}
-				}
+				// Restart rangefeed on another goroutine. Restart might be a bit
+				// expensive, particularly if we have to resolve span.  We do not want
+				// to block receiveEventsFromNode for too long.
+				toRestart := ms.purgeStuckStreams(threshold)
+				m.g.GoCtx(func(ctx context.Context) error {
```

Comment:
Could maybe skip spinning up a goroutine if `len(toRestart) == 0` (which should be the common fast-path?).

Reply:
Definitely.